src/char/int_storage.c

```diff
@@ -41,24 +41,25 @@ struct inter_storage_interface inter_storage_s;
 struct inter_storage_interface *inter_storage;
 
 /// Save storage data to sql
-int inter_storage_tosql(int account_id, struct storage_data* p)
+int inter_storage_tosql(int account_id, const struct item *p)
```

Comment:
I think it's better for this function to stay as it was (and just add const). Similar things hurt us in the past, and I don't want to repeat it (it's the reason why I started using VECTOR wherever we were using pointers to the first element of an array)

`struct storage_data *` has the benefit over `struct item *` of containing an array of known size. If you really want to pass a pointer to the first entry, then you have to always pass the size along with it.


The following are both equally acceptable:
```
// This passes the knowledge that the item array is part of a storage, and as such the function knows that, by definition, it'll have MAX_STORAGE elements
int inter_storage_tosql(int account_id, const struct storage_data* p)

// This pushes the responsibility on the caller, which knows that the item array is part of a storage, of letting the function know the array size
int inter_storage_tosql(int account_id, const struct item *p, int length)
```

I don't want functions that take a pointer to the first element of an array, and arbitrarily decide the size of said array.